spec/ledger_spec.rb

```diff
@@ -0,0 +1,28 @@
+require './lib/ledger.rb'
+
+describe Ledger do
+  subject(:my_ledger) { described_class.new }
+
+  describe '#attributes' do
+    it 'initializes with an empty transactions array' do
+      expect(my_ledger.transactions).to eq([])
+    end
+    it 'initializes with a balance of 0' do
+      expect(my_ledger.balance).to eq(0)
+    end
+  end
+
+  describe '#print_statement' do
+    it 'responds to the print_statement method' do
+      expect(my_ledger).to respond_to(:print_statement)
+    end
+    it 'allows me to print a full statement' do
+      my_ledger.transactions = [{ amount: 100, date: '07/08/2017', type: :credit }]
```

Comment:
This test feels like it relies on the fact that the the transactions (which are implementation details) are not encapsulated.  Is there a way to write this test that just uses the public behaviour of the class?